server/id/id_impl.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package id
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"strconv"
+	"sync"
+
+	"github.com/CeresDB/ceresmeta/pkg/log"
+	"github.com/CeresDB/ceresmeta/server/storage"
+	"github.com/pkg/errors"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.uber.org/zap"
+)
+
+const defaultAllocStep = uint64(1000)
+
+type AllocatorImpl struct {
+	// lock var base&end when allocator ID
+	sync.Mutex
+	base     uint64
+	end      uint64
+	kv       storage.KV
+	rootPath string
+	key      string
+}
+
+func NewAllocatorImpl(kv storage.KV, rootPath string, key string) *AllocatorImpl {
+	return &AllocatorImpl{kv: kv, key: key, rootPath: rootPath}
+}
+
+func (alloc *AllocatorImpl) Alloc(ctx context.Context) (uint64, error) {
+	alloc.Lock()
+	defer alloc.Unlock()
+
+	if alloc.base == alloc.end {
+		if err := alloc.fastRebaseLocked(ctx); err != nil {
+			if err := alloc.rebaseLocked(ctx); err != nil {
+				return 0, err
+			}
+		}
+	}
+
+	alloc.base++
+	return alloc.base, nil
+}
+
+func (alloc *AllocatorImpl) rebaseLocked(ctx context.Context) error {
+	value, err := alloc.kv.Get(ctx, alloc.key)
+	if err != nil {
+		return errors.Wrapf(err, "get base id failed, key:%v", alloc.key)
+	}
+	if value != "" {
+		value, err := strconv.ParseUint(value, 10, 64)
```

Comment:
It is better to define a method called `decodeID` which corresponds to the `encodeID`.